x/storage/keeper/keeper.go

```diff
@@ -1690,6 +1753,44 @@ func (k Keeper) garbageCollectionForResource(ctx sdk.Context, deleteStalePolicie
 	return deletedTotal, true
 }
 
+func (k Keeper) MigrationBucket(ctx sdk.Context, srcSP, dstSP *sptypes.StorageProvider, bucketInfo *types.BucketInfo) error {
+	store := ctx.KVStore(k.storeKey)
+
+	key := types.GetMigrationBucketKey(bucketInfo.Id)
+	if store.Has(key) {
+		panic("migration bucket key is existed.")
```

Comment:
I think user may migrate a bucket twice, no need panic here.

Reply:
We will add cancel migration bucket API. Therefore, no migration is allowed during the migration